Experimental/src/org/sleuthkit/autopsy/experimental/autoingest/DeleteCaseTask.java

```diff
@@ -0,0 +1,850 @@
+                        }
+                    } catch (InterruptedException ex) {
+                        logger.log(Level.WARNING, String.format("Deletion of %s cancelled while incomplete", caseNodeData.getDisplayName()), ex);
+                        releaseManifestFileLocks();
+                        return;
+                    }
+
+                    if (Thread.currentThread().isInterrupted()) {
+                        logger.log(Level.WARNING, String.format("Deletion of %s cancelled while incomplete", caseNodeData.getDisplayName()));
+                        releaseManifestFileLocks();
+                        return;
+                    }
+
+                    progress.progress(Bundle.DeleteCaseTask_progress_deletingJobLogLockNode());
+                    logger.log(Level.INFO, String.format("Deleting case auto ingest job log znode for %s", caseNodeData.getDisplayName()));
+                    String logFilePath = CoordinationServiceUtils.getCaseAutoIngestLogNodePath(caseNodeData.getDirectory());
+                    try {
+                        coordinationService.deleteNode(CategoryNode.CASES, logFilePath);
+                    } catch (CoordinationServiceException ex) {
+                        if (!isNoNodeException(ex)) {
+                            logger.log(Level.SEVERE, String.format("Error deleting case auto ingest job log znode for %s", caseNodeData.getDisplayName()), ex);
+                        }
+                    } catch (InterruptedException ex) {
+                        logger.log(Level.WARNING, String.format("Deletion of %s cancelled while incomplete", caseNodeData.getDisplayName()), ex);
+                        releaseManifestFileLocks();
+                        return;
+                    }
+                }
+
+                if (Thread.currentThread().isInterrupted()) {
+                    logger.log(Level.WARNING, String.format("Deletion of %s cancelled while incomplete", caseNodeData.getDisplayName()));
+                    releaseManifestFileLocks();
+                    return;
+                }
+
+                if (deleteOption == DeleteOptions.DELETE_OUTPUT || deleteOption == DeleteOptions.DELETE_ALL) {
+                    try {
+                        deleteManifestFileNodes();
+                    } catch (InterruptedException ex) {
+                        logger.log(Level.WARNING, String.format("Deletion of %s cancelled while incomplete", caseNodeData.getDisplayName()), ex);
+                        return;
+                    }
+                }
+
+                releaseManifestFileLocks();
+
+            } catch (CoordinationServiceException ex) {
+                logger.log(Level.SEVERE, String.format("Could not delete %s because an error occurred acquiring the case directory lock", caseNodeData.getDisplayName()), ex);
+                return;
```

Comment:
Should the manifest file locks be released here?

Reply:
See discussion in Jira.